Blocstagram/Blocstagram/DataSource.m

```diff
@@ -0,0 +1,112 @@
+//
+//  DataSource.m
+//  Blocstagram
+//
+//  Created by Weinan Qiu on 2015-08-25.
+//  Copyright (c) 2015 Bloc. All rights reserved.
+//
+
+#import "DataSource.h"
+#import "User.h"
+#import "Media.h"
+#import "Comment.h"
+
+@interface DataSource ()
+
+@property (nonatomic, strong) NSArray *mediaItems;
+
+@end
+
+@implementation DataSource
+
++ (instancetype)sharedInstance {
+    static dispatch_once_t once;
+    static id sharedInstance;
+    
+    dispatch_once(&once, ^{
+        sharedInstance = [[self alloc] init];
+    });
+    
+    return sharedInstance;
+}
+
+- (instancetype)init {
+    self = [super init];
+    if (self) {
+        [self addRandomData];
+    }
+    return self;
+}
+
+- (void)addRandomData {
+    NSMutableArray *randomMediaItems = [NSMutableArray array];
+    
+    for (int i = 1; i <= 10; i++) {
+        UIImage *image = [UIImage imageNamed:[NSString stringWithFormat:@"%d.jpg", i]];
```

Comment:
As another part of my debugging, I also added a log where we were loading the image, to make sure maybe our images weren't being corrupted when we loaded them:
```
        NSLog(@"Loading images... image %d -- %@", i, image);
```
The image description that gets printed out gives the memory address of the image and the height and width, like so:
```
2015-08-31 10:20:44.592 Blocstagram[20234:4962172] Loading images... image 1 -- <UIImage: 0x7f8d024cbcf0>, {2112, 2816}
```

And this was a quick way to rule out corrupted images.